source/gameengine/GameLogic/SCA_VibrationActuator.cpp

```diff
@@ -62,16 +64,32 @@ bool SCA_VibrationActuator::Update()
 		return false;
 	}
 
-	bool bPositiveEvent = IsPositiveEvent();
+	if (IsPositiveEvent()) {
+		switch (m_mode) {
+			case KX_ACT_VIBRATION_PLAY:
+			{
+				instance->RumblePlay(m_strength, m_strength_right, m_duration);
+				m_endtime = PIL_check_seconds_timer() * 1000.0f + m_duration;
+				break;
+			}
+			case KX_ACT_VIBRATION_STOP:
+			{
+				instance->RumbleStop();
+				m_endtime = 0.0f;
+				break;
+			}
+		}
 
-	if (bPositiveEvent) {
-		instance->RumblePlay(m_strength, m_duration);
-		m_endtime = PIL_check_seconds_timer() * 1000.0f + m_duration;
 	}
 
 	RemoveAllEvents();
 
-	return PIL_check_seconds_timer() * 1000.0f < m_endtime;
+	if (!(PIL_check_seconds_timer() * 1000.0f < m_endtime)) {
+		instance->RumbleStop();
+		m_endtime = 0.0f;
+	}
+
+	return instance->GetRumbleStatus();
```

Comment:
Hi,

the m_endtime should not be the same in different actuators. It is an internal count for playing and it fulfil its task.

Regarding to the return behaviour if the joystick supports or not vibration I think that it should be different as it follows the reality. 

BUT you are pointing well out a weak point. To mitigate this weak i have included a new "hasVibration" attribute to can check if the joystick supports or not vibration. 

Thanks.